lib/Runtime/Library/JavascriptLibrary.cpp

```diff
@@ -1116,12 +1116,15 @@ namespace Js
         AddFunctionToLibraryObject(globalObject, PropertyIds::escape, &GlobalObject::EntryInfo::Escape, 1);
         AddFunctionToLibraryObject(globalObject, PropertyIds::unescape, &GlobalObject::EntryInfo::UnEscape, 1);
 
-        if (scriptContext->GetConfig()->SupportsCollectGarbage()
+// for backward compat reasons in non-core builds add CollectGarbage function even if it would do nothing later
+#ifdef _CHAKRACOREBUILD
```

Comment:
>_CHAKRACOREBUILD [](start = 7, length = 16)

Why restrict to ChakraCore only? Sometimes CollectGarbage is useful in full build to repro inconsistent bugs.

Reply:
In ChakraCore we add CollectGarbage to the globals based on IsCollectGarbageEnabled, otherwise we add it unconditionally. So this change isn't removing any functionality from the full builds, but would still need to pass in the flag for GlobalObject::EntryCollectGarbage to both register and actually do garbage collection.

---
In reply to: [178200976](https://github.com/Microsoft/ChakraCore/pull/4909#discussion_r178200976) [](ancestors = 178200976)